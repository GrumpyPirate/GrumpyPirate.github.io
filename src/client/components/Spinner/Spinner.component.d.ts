interface SpinnerProps {
  isLoading?: boolean;
}

export { SpinnerProps };
