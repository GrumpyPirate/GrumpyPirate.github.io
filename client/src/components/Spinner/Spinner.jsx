import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'components/Icon/Icon';

import classes from './Spinner.scss';

const Spinner = ({ isLoading }) => (
  <div
    className={classnames(classes['spinner'], {
      [classes['spinner--is-loading']]: isLoading,
    })}
  >
    <figure className={classes['spinner__icon']}>
      <Icon glyph="loader" altText="Loading..." />
    </figure>

    <h6 className={classes['spinner__loading-text']}>Loading...</h6>
  </div>
);

Spinner.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Spinner;
