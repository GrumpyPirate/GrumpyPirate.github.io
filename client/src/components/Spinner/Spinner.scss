@import 'core';

@keyframes spin {
  from { transform: rotateZ(0turn); }
  to { transform: rotateZ(1turn); }
}

.spinner {
  align-items: center;
  background-color: palette('theme-mid');
  bottom: 0;
  color: palette('theme-light');
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  left: 0;
  opacity: 0;
  pointer-events: none;
  position: fixed;
  right: 0;
  top: $sidebar-height-mobile;
  transition-duration: $anim-dur * 2;
  transition-property: opacity, visibility;
  transition-timing-function: $anim-ease;
  user-select: none;
  visibility: hidden;
  z-index: 2;

  @include respond-up('lg') {
    left: $sidebar-width-lg;
    top: 0;
  }

  @include respond-up('xl') {
    left: $sidebar-width;
  }

  &__icon {
    margin: 0 0 $paragraph-margin-bottom;

    svg {
      animation: spin ($anim-dur * 15) linear 0s infinite;
      display: block;
      height: 3rem;
      margin: 0 auto;
      width: 3rem;
    }
  }

  &__loading-text {
    @include heading(4);
    left: rem(6px);
    margin: 0;
    position: relative;
    text-transform: uppercase;
  }

  &--is-loading {
    opacity: 1;
    visibility: visible;
  }
}
