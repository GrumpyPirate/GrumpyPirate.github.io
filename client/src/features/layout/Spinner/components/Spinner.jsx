import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'components/Icon/Icon';

import './Spinner.scss';

const Spinner = ({ isLoading }) => (
  <div
    className={classnames('spinner', {
      'spinner--is-loading': isLoading,
    })}
  >
    <figure className="spinner__icon">
      <Icon glyph="loader" altText="Loading..." />
    </figure>

    <h6 className="h4 mb-0">Loading...</h6>
  </div>
);

Spinner.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Spinner;
