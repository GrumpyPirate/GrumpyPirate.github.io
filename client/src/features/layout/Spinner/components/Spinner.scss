@import 'core';

.spinner {
  position: fixed;
  top: $sidebar-height-mobile;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
  background-color: palette(theme-mid);
  color: palette(theme-light);
  z-index: 2;
  opacity: 0;
  visibility: hidden;
  pointer-events: none;
  user-select: none;
  transition: opacity ($anim-dur * 2) $anim-ease,
              visibility ($anim-dur * 2) $anim-ease;

  // lg up
  @include media-breakpoint-up(lg) {
    top: 0;
    left: $sidebar-width;
  }

  &__icon {
    .icon {
      display: block;
      width: 3rem;
      height: 3rem;
      margin: 0 auto;
      animation: loader ($anim-dur * 15) linear 0s infinite forwards;
    }
  }

  &--is-loading {
    opacity: 1;
    visibility: visible;
  }
}
