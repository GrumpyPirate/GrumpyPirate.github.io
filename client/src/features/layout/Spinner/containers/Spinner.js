import { connect } from 'react-redux'

import Spinner from '../components/Spinner';

const mapStateToProps = state => ({
  isLoading: state.portfolio.isFetching
})

export default connect(mapStateToProps)(Spinner)
