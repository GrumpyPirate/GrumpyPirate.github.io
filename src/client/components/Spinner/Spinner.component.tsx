import React, { FunctionComponent } from 'react';
import classnames from 'classnames';

import Icon from 'components/Icon/Icon.component';

import { SpinnerProps } from './Spinner.component.d';
import classes from './Spinner.component.scss';

const Spinner: FunctionComponent<SpinnerProps> = ({ isLoading = false }: SpinnerProps) => (
  <div
    className={classnames(classes['spinner'], {
      [classes['spinner--is-loading']]: isLoading,
    })}
  >
    <figure className={classes['spinner__icon']}>
      <Icon glyph="loader" altText="Loading..." />
    </figure>

    <h6 className={classes['spinner__loading-text']}>Loading...</h6>
  </div>
);
export default Spinner;
