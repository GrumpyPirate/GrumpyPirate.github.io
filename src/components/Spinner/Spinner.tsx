import React, { FunctionComponent } from 'react';

import Icon from 'components/Icon/Icon';

import { Image, Text, Wrapper } from './Spinner.constants';

const Spinner: FunctionComponent = () => (
  <Wrapper>
    <Image>
      <Icon glyph="loader" altText="Loading..." />
    </Image>

    <Text>Loading...</Text>
  </Wrapper>
);

export default Spinner;
