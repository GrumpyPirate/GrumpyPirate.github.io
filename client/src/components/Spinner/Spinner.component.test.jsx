import React from 'react';
import { shallow } from 'enzyme';

import Spinner from './Spinner.component';

import fixture from './Spinner.component.fixture';

describe('Components', () => {
  describe('Spinner', () => {
    const props = {
      ...fixture,
    };

    it('should be defined', () => {
      expect(Spinner).toBeDefined();
    });

    it('should be a function', () => {
      expect(typeof Spinner).toBe('function');
    });

    it('should render as expected, without crashing', () => {
      const wrapper = shallow(<Spinner {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
