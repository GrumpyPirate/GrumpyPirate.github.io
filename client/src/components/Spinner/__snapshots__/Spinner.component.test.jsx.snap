// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Spinner should render as expected, without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner
    isLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <figure
          className="spinner__icon"
        >
          <Icon
            altText="Loading..."
            className="icon"
            glyph="loader"
          />
        </figure>,
        <h6
          className="spinner__loading-text"
        >
          Loading...
        </h6>,
      ],
      "className": "spinner spinner--is-loading",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Icon
            altText="Loading..."
            className="icon"
            glyph="loader"
          />,
          "className": "spinner__icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "altText": "Loading...",
            "className": "icon",
            "glyph": "loader",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "figure",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "spinner__loading-text",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "h6",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <figure
            className="spinner__icon"
          >
            <Icon
              altText="Loading..."
              className="icon"
              glyph="loader"
            />
          </figure>,
          <h6
            className="spinner__loading-text"
          >
            Loading...
          </h6>,
        ],
        "className": "spinner spinner--is-loading",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Icon
              altText="Loading..."
              className="icon"
              glyph="loader"
            />,
            "className": "spinner__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "altText": "Loading...",
              "className": "icon",
              "glyph": "loader",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "figure",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "spinner__loading-text",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "h6",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
